src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs

```diff
@@ -0,0 +1,195 @@
+            int minDataPerLeaf = Args.MinDataPerLeaf ?? DefaultMinDataPerLeaf(numRow, numLeaves, _numClass);
+            Options["min_data_per_leaf"] = minDataPerLeaf.ToString();
+            if (!hiddenMsg)
+            {
+                if (!Args.LearningRate.HasValue)
+                    ch.Info("Auto-tuning parameters: " + nameof(Args.LearningRate) + " = " + Options["learning_rate"]);
+                if (!Args.NumLeaves.HasValue)
+                    ch.Info("Auto-tuning parameters: " + nameof(Args.NumLeaves) + " = " + numLeaves);
+                if (!Args.MinDataPerLeaf.HasValue)
+                    ch.Info("Auto-tuning parameters: " + nameof(Args.MinDataPerLeaf) + " = " + minDataPerLeaf);
+            }
+        }
+
+        protected override void CheckAndUpdateParametersBeforeTraining(IChannel ch, RoleMappedData data, float[] labels, int[] groups)
+        {
+            Host.AssertValue(ch);
+            ch.Assert(PredictionKind == PredictionKind.MultiClassClassification);
+            ch.Assert(_numClass > 1);
+            Options["num_class"] = _numClass.ToString();
+            bool useSoftmax = false;
+
+            if (Args.UseSoftmax.HasValue)
+                useSoftmax = Args.UseSoftmax.Value;
+            else
+            {
+                if (labels.Length >= _minDataToUseSoftmax)
+                    useSoftmax = true;
+
+                ch.Info("Auto-tuning parameters: " + nameof(Args.UseSoftmax) + " = " + useSoftmax);
+            }
+
+            if (useSoftmax)
+                Options["objective"] = "multiclass";
+            else
+                Options["objective"] = "multiclassova";
+
+            // Add default metric.
+            if (!Options.ContainsKey("metric"))
+                Options["metric"] = "multi_error";
+        }
+    }
+
+    /// <summary>
+    /// A component to train an LightGBM model.
+    /// </summary>
+    public static partial class LightGbm
+    {
+        [TlcModule.EntryPoint(Name = "Trainers.LightGbmClassifier", Desc = "Train an LightGBM multi class model", UserName = LightGbmMulticlassTrainer.Summary, ShortName = LightGbmMulticlassTrainer.ShortName)]
+        public static CommonOutputs.MulticlassClassificationOutput TrainMultiClass(IHostEnvironment env, LightGbmArguments input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register("TrainLightGBM");
+            host.CheckValue(input, nameof(input));
+            EntryPointUtils.CheckInputArgs(host, input);
+
+            return LearnerEntryPointsUtils.Train<LightGbmArguments, CommonOutputs.MulticlassClassificationOutput>(host, input,
+                () => new LightGbmMulticlassTrainer(host, input),
+                getLabel: () => LearnerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.LabelColumn),
+                getWeight: () => LearnerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.WeightColumn));
```

Comment:
>getWeight [](start = 16, length = 9)

add getGroup:

Reply:
I'm not sure if there is a need to.

---
In reply to: [197550320](https://github.com/dotnet/machinelearning/pull/392#discussion_r197550320) [](ancestors = 197550320)